storage/s3.go

```diff
@@ -555,6 +555,13 @@ func (s *S3) Copy(ctx context.Context, from, to *url.URL, metadata Metadata) err
 		input.MetadataDirective = aws.String(metadata.Directive)
 	}
 
+	if metadata.ContentType != "" {
+		input.ContentType = aws.String(metadata.ContentType)
+		if metadata.Directive == "" {
+			input.MetadataDirective = aws.String("REPLACE")
+		}
+	}
+
```

Comment:
```suggestion
	if metadata.ContentType != "" {
		input.ContentType = aws.String(metadata.ContentType)
	}
	
	if metadata.Directive == "" {
	         input.MetadataDirective = aws.String("REPLACE")
	}
```
Two of them are independent. One's value should not effect the other for the request. `MetadataDirective` can be set even when `ContentType` is empty.

Reply:
Actually, there is no need for a second condition because this code block already covers this case.

```go
  if c.metadataDirective == "" {
          // default to COPY
          c.metadataDirective = "COPY"
          if c.src.IsRemote() && c.dst.IsRemote() && c.contentType != "" {
                  // default to REPLACE for content type change
                  c.metadataDirective = "REPLACE"
          }
  }
```

Reply:
It's very confusing to follow. The nested if's can be flattened, it's easier for reading&understanding. Also, the code block checks only if the `contentType` is not empty but the `metadataDirective` affects all metadata related fields, `contentDisposition, metadata` etc. If it's going to be set `REPLACE` when `s3-to-s3` copy will be performed, why `contentType` is only metadata related field to be checked ? It looks like we shouldn't decide based on the `contentType` but only to check if the transfer is `s3-to-s3`. Also, the directive should be set above of all metadata related fields, since it affects those fields, regarding the code block in `storage/s3.go`(L558 and forward). Also if the `command/cp.go` sets the default value, why do we need to set a value to `metadataDirective` in `storage/s3.go` ?

Reply:
Sounds right 👍 The tests should cover the second point: 
>In the suggestion change code you wrote, only the first condition is necessary. This is because the meta-directive will already be determined before reaching this point.

Reply:
done.